src/test/util.rs

```diff
@@ -131,6 +119,16 @@ impl TestClientBuilder {
             options.test_options_mut().min_heartbeat_freq = Some(freq);
         }
 
+        if self.sharded_use_first_host {
```

Comment:
Deferring the action to here means there's no issue with order of calls with setting options and it avoids the need for an additional `.await` in the builder call chain.